simulator/src/main/java/com/hazelcast/simulator/common/SimulatorProperties.java

```diff
@@ -16,82 +16,61 @@
  */
 public class SimulatorProperties {
+
+    static final String PROPERTIES_FILE_NAME = "simulator.properties";
+
     private static final Logger LOGGER = Logger.getLogger(SimulatorProperties.class);
 
     private final Properties properties = new Properties();
+
     private String forcedHazelcastVersionSpec;
 
     public SimulatorProperties() {
-        File defaultPropsFile = newFile(getSimulatorHome(), "conf", "simulator.properties");
-        LOGGER.debug("Loading default simulator.properties from: " + defaultPropsFile.getAbsolutePath());
+        File defaultPropsFile = newFile(getSimulatorHome(), "conf", PROPERTIES_FILE_NAME);
+        LOGGER.debug(format("Loading default %s from: %s", PROPERTIES_FILE_NAME, defaultPropsFile.getAbsolutePath()));
+        check(defaultPropsFile);
         load(defaultPropsFile);
     }
 
-    public String getUser() {
-        return get("USER", "simulator");
-    }
-
-    public boolean isEc2() {
-        return "aws-ec2".equals(get("CLOUD_PROVIDER"));
-    }
-
-    public String getHazelcastVersionSpec() {
-        if (forcedHazelcastVersionSpec == null) {
-            return get("HAZELCAST_VERSION_SPEC", "outofthebox");
-        } else {
-            return forcedHazelcastVersionSpec;
-        }
-    }
-
-    public void forceGit(String gitRevision) {
-        if (gitRevision != null && !gitRevision.isEmpty()) {
-            forcedHazelcastVersionSpec = HazelcastJars.GIT_VERSION_PREFIX + gitRevision;
-            LOGGER.info("Overriding Hazelcast version to GIT revision " + gitRevision);
-        }
-    }
-
     /**
-     * Initialized the SimulatorProperties
+     * Initializes the SimulatorProperties.
      *
      * @param file the file to load the properties from. If the file is null, then first the simulator.properties
      *             in the working dir is checked.
      */
     public void init(File file) {
         if (file == null) {
-            //if no file is explicitly given, we look in the working directory
-            File fallbackPropsFile = new File("simulator.properties");
-            if (fallbackPropsFile.exists()) {
-                file = fallbackPropsFile;
-            } else {
-                LOGGER.warn(format("%s is not found, relying on defaults", fallbackPropsFile));
+            // if no file is explicitly given, we look in the working directory
+            file = new File(PROPERTIES_FILE_NAME);
+            if (!file.exists()) {
+                LOGGER.warn(format("%s is not found, relying on defaults", file.getAbsolutePath()));
+                return;
             }
         }
 
-        if (file != null) {
-            LOGGER.info(format("Loading simulator.properties: %s", file.getAbsolutePath()));
-            load(file);
-        }
+        LOGGER.info(format("Loading %s: %s", PROPERTIES_FILE_NAME, file.getAbsolutePath()));
+        check(file);
+        load(file);
     }
 
-    private void load(File file) {
+    private void check(File file) {
```

Comment:
I pulled out this check so I could write tests for the `load()` method for more code branches.